gnssanalysis/gn_io/sp3.py

```diff
@@ -365,7 +395,38 @@ def clk_std_formatter(x):
             f" {epoch_datetime.hour:2} {epoch_datetime.minute:2} {frac_seconds:11.8f}"
         )
         out_buf.write("\n")
+
         # Format this epoch's values in the SP3 format and write to buffer
+
+        # First, we fill NaN and infinity values with the standardised nodata value for each column.
+        # NOTE: DataFrame.to_string() as called below, takes formatter functions per column. It does not, however
+        # invoke them on NaN values!! As such, trying to handle NaNs in the formatter is a fool's errand.
+        # Instead, we do it here, and get the formatters to recognise and skip over the already processed nodata values
+        
+        # POS nodata formatting
```

Comment:
Awesome, this whole section reads well - pretty clear what is being done and in a logical manner 👍🏻